auspice/client/languageSelector.js

```diff
@@ -0,0 +1,78 @@
+import React from "react"; // eslint-disable-line
+
+
+class LanguageSelector extends React.Component {
+    state = {
+      currentNarrative: undefined,
+      availableNarratives: undefined
+    }
+
+    async getAvailableNarratives() {
+      const response = await fetch("/charon/getAvailable");
+      const available = await response.json();
+      return available.narratives.map((n) => n.request);
+    }
+
+    getNarrativeLanguage(narrative) {
+      const narrativeParts = narrative.split("/");
+      let narrativeLanguage = narrativeParts[narrativeParts.length - 2];
+      const languageMap = {
+        en: "English",
+        zh: "Chinese",
+        es: "Spanish",
+        pt: "Portuguese",
+        de: "German"
```

Comment:
Best practice is to present language names in the language itself, and this is what the existing translated narrative links do too.